streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -1496,15 +1553,78 @@ public void shouldRecoverFromInvalidOffsetExceptionOnRestoreAndFinishRestore() t
     }
 
     @Test
-    public void shouldRecordSkippedMetricForDeserializationException() {
+    public void shouldRecordSkippedMetricForDeserializationExceptionWithBuiltInMetricsVersionLatest() {
```

Comment:
nit: shouldRecordSkippedMetric -> shouldLog? With latest we do not record skipped metric, but the dropped-records and logging right?

Reply:
Also I'm wondering if we could consolidate this with the `shouldRecordSkippedMetricForDeserializationExceptionWithBuiltInMetricsVersion0100To23` below: based on the streams metrics config, only the validation is different. Similar to the test cases above.

Reply:
Good point! Consolidating the two tests is better.